roles/nvidia-gpu-operator/tasks/main.yml

```diff
@@ -27,7 +26,24 @@
     gpu_operator_registry_secret: --set driver.imagePullSecrets[0]="{{ gpu_operator_registry_secret }}"
   when: gpu_operator_registry_secret != ""
 
+- name: Ensure config dir exists
+  file:
+    path: "{{ gpu_operator_config_dir }}"
+    state: directory
+    owner: root
+    group: root
+
+- name: Copy DCGM CSV config file to host
+  copy:
+    src: "{{ gpu_operator_dcgm_config_csv }}"
+    dest: "{{ gpu_operator_config_dir }}/{{ gpu_operator_dcgm_config_csv }}"
+
+- name: Copy values.yml to host
+  template:
+    src: "{{ gpu_operator_values_file }}"
+    dest: "{{ gpu_operator_config_dir }}/{{ gpu_operator_values_file }}"
+
 # XXX: This currently installs into the default namespace, as per the GPU Operator docs
 # https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/getting-started.html
 - name: install nvidia gpu operator
-  command: /usr/local/bin/helm upgrade --install "{{ gpu_operator_release_name }}" "{{ gpu_operator_chart_name }}" --version "{{ gpu_operator_chart_version }}" --set migStrategy="{{ k8s_gpu_mig_strategy }}" --set operator.defaultRuntime="{{ gpu_operator_default_runtime }}" --set driver.repository="{{ gpu_operator_driver_registry }}" --set driver.version="{{ gpu_operator_driver_version }}" {{ gpu_operator_registry_secret }} --set gfd.migStrateg="{{ k8s_gpu_mig_strategy }}" --set devicePlugin.args={"{{ gpu_operator_plugin_args }}"} --wait
+  command: /usr/local/bin/helm upgrade --install "{{ gpu_operator_release_name }}" "{{ gpu_operator_chart_name }}" --version "{{ gpu_operator_chart_version }}" -f "{{ gpu_operator_config_dir }}/{{ gpu_operator_values_file }}"
```

Comment:
Ah, I see. You just went ahead and used the temp config. This is cleaner now that we have so much junk we are tacking on here.

Reply:
Actually, I take that back. It looks like you dropped `gpu_operator_registry_secret` and this will cause the vGPU installation to fail.